### PR TITLE
Implemented conv1d support

### PR DESCRIPTION
**Pull request by borisfom**
Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

# Description

Added support for 1d convolutions.
Also:
Fixed 1d case for batch normalization
Included a fix for prim::ListConstruct when arg is a Tuple (apparently valid use case)
Fixes # (issue)

## Type of change
- New feature (non-breaking change which adds functionality)
- 
# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ x] I have added tests to verify my fix or my feature
- [ x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/converters/impl/batch_norm.cpp [hunk 1]
```diff
@@ -65,14 +58,12 @@ auto batch_norm_registrations TRTORCH_UNUSED = RegisterNodeConversionPatterns().
...
-      if (should_unpack) {
-        LOG_DEBUG("Inserting shuffle layer to reshape to back to original shape: " << orig_shape);
-        auto out_shuffle = ctx->net->addShuffle(*out_tensor);
-        out_shuffle->setReshapeDimensions(orig_shape);
-        out_shuffle->setName(std::string("[Reshape output to " + util::toStr(orig_shape) + ']').c_str());
-        out_tensor = out_shuffle->getOutput(0);
+      if (expandDims) {
+          LOG_DEBUG("Inserting shuffle layer to reshape to back to original shape: " << orig_shape);
+          auto new_layer = util::unpadTensorDim(ctx, n, out_tensor, orig_shape.nbDims);
+          assert(new_layer);
```
Comment (peri044) [inline core/conversion/converters/impl/batch_norm.cpp:64]: We can use `TRTORCH_CHECK` here

File: core/conversion/converters/impl/conv_deconv.cpp [hunk 2]
```diff
@@ -9,34 +9,68 @@ namespace conversion {
...
+      in = tensorPtr;
+      stride=util::unsqueezeDims(stride, 1, 1);
+      dilation=util::unsqueezeDims(dilation, 1, 1);
+      padding=util::unsqueezeDims(padding, 1, 0);
+      out_padding=util::unsqueezeDims(out_padding, 1, 0);
+    }
+  if (w.shape.nbDims < 4)
+    {
+      for (int i = w.shape.nbDims; i <4 ; ++i) 
+	w.shape.d[i] = 1;
```
Comment (peri044) [inline core/conversion/converters/impl/conv_deconv.cpp:45]: probably the formatting here will the fixed by linter ?

**Review by peri044 — CHANGES_REQUESTED** [review]

File: core/conversion/converters/converter_util.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#include "core/conversion/converters/Weights.h"
+#include "core/conversion/var/Var.h"
+#include "core/util/prelude.h"
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+namespace util {
+    using namespace core::util;
```
[the remark this answers is not in the document]
Comment (borisfom) [reply core/conversion/converters/converter_util.h:16]: If I remove this, then all existing util:: references in converters code would have to be changed to core::util::.

File: core/conversion/converters/converter_util.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  nvinfer1::ILayer* padTensorDim(ConversionCtx* ctx, const torch::jit::Node* n, nvinfer1::ITensor* tensor, int nDim)
+    {
+        const auto dims = tensor->getDimensions();
+	
+        if (dims.nbDims < nDim)
+        {
+	    auto newDims = dims;
+	    for (int dim = dims.nbDims; dim < nDim; ++dim)
+	      newDims = unsqueezeDims(newDims, dim);
```
[the remark this answers is not in the document]
Comment (borisfom) [reply core/conversion/converters/converter_util.cpp:19]: Not sure what that would buy us. Anyway, probably, it's better to not have converters::util namespace at all.

File: core/conversion/converters/converter_util.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#include "core/util/prelude.h"
+#include "core/conversion/converters/converter_util.h"
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+namespace util {
+
+  nvinfer1::ILayer* padTensorDim(ConversionCtx* ctx, const torch::jit::Node* n, nvinfer1::ITensor* tensor, int nDim)
```
[the remark this answers is not in the document]
Comment (borisfom) [reply core/conversion/converters/converter_util.cpp:11]: Sure, it would make sense to use name that would hint at instantiating a new layer :).

File: core/conversion/converters/converter_util.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (peri044) [reply core/conversion/converters/converter_util.h:16]: We usually use `util::` for all utility operations (like unPadDims, toVec etc) when we use them in converters. We also try to avoid using `using namespaces` generally.

**Comment by borisfom** [comment]
Superseded by https://github.com/NVIDIA/TRTorch/pull/435
